tests/test_models.py

```diff
@@ -202,56 +237,158 @@ def tearDown(self):
         self.mail = None
 
 
-class MailTemplateTest(TestCase):
+@override_settings(SENDGRID_API_KEY="test-api-key")
+class MailAttachmentsSendTest(TestCase):
+    file_contents = b"Here the file contents"
 
+    def setUp(self):
+        self.mail_template = MailTemplate(
+            name="test_template",
+            subject="Message for {{ name }}",
+            html_template="<p><b>Hello</b>, {{ name }}!</p>",
+        )
+        self.mail_template.save()
+        self.context = Context({"name": "Max"})
+        self.media_root_old = settings.MEDIA_ROOT
+        self.temp_dir = pathlib.Path(tempfile.mkdtemp())
+        settings.MEDIA_ROOT = self.temp_dir
+        self.attachment = Attachment(
+            name="attachment1",
+            attached_file=SimpleUploadedFile("input.txt", self.file_contents),
+        )
+        self.attachment.save()
+        TemplateAttachment.objects.create(
+            template=self.mail_template, attachment=self.attachment
+        )
+
+    @mock.patch.object(Client, "post", create=True)
+    def testGridMailSend(self, post):
+        self.mail = Mail.objects.create_mail(
+            self.mail_template,
+            {"name": "Max"},
+            "mailtest@sink.sendgrid.net",
+            "test@example.com",
+            subject="Custom subject",
+            attachments=[self.attachment],
+        )
+
+        # Send the mail and record time sent
+        self.mail.send(sendgrid_api=True)
+        args, kwargs = post.call_args
+        content = base64.b64decode(kwargs["request_body"]["attachments"][0]["content"])
+        self.assertEqual(content, self.file_contents)
+
+    def testMailSend(self):
```

Comment:
It would be good to have another test to show that it works without any attachments.

Reply:
Is there any difference between file that is uploaded before and file that is uploaded now?
Some attachments are related to single mail only, and we probably need to make some of them available for templates. In that case maybe a boolean field `shareable` could be added to attachment model (True = can be selected for adding to templates) instead of creating another model (UploadedMailAttachment)

Reply:
> Some attachments are related to single mail only, and we probably need to make some of them available for templates. In that case, maybe a boolean field shareable could be added to the attachment model (True = can be selected for adding to templates) instead of creating another model (UploadedMailAttachment)

You are right. This is to be discussed. We could use an `AbstractAttachment` and reuse the most attributes. But still, the logic behind those different attachments is a little different and its good to separate those. 

Template Attachments are like 10-15 files used in some templates. UploadedMailAttachments would be way more as this could be 0-* files for every sent mail. 

But: let's deal with this later - IF we want to add attachments individually for a mail. 

And for the test, it was meant, that we just make sure, that there is no relation in any way between a Template, its attachments, and the mail. If you send a mail with 0 attachments, we are not using any attachments of the Template.

Reply:
Okay, I understand. Added `AbstractAttachment` model and subclassed Attachment from it.